examples/systemcalls/systemcalls.c

```diff
@@ -93,7 +138,58 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *
 */
 
+	int fd = open(outputfile, O_WRONLY | O_TRUNC | O_CREAT , 0644);
+	if(fd < 0)
+	{
+		abort();
+	}
+	
+	fflush(stdout);	//avoid duplicate prints
+	int status;
+	pid_t pid;
+	
+	pid = fork();
```

Comment:
@JayashCUB how could you share logic with the function above and avoid duplication in these two functions?